### ISSUE TITLE
updateRecall failing due to array size

### ISSUE THREAD

**Issue by barrelltech** [fasiha/ebisu.js#15]
Hello - 

I love the idea behind Ebisu and would like to implement it in a project of mine. However, whenever I attempt to `updateRecall` - with any value - I'm running into the error:

```
RangeError: Array size is not a small enough positive integer
```

I'm not given any stack trace (into Ebisu itself, just that it's happening at the invocation line). For example. some of the calls I'm running are

```
ebisu.updateRecall([4, 4, 30000], 0.6077981651376146, 1, 1072)
ebisu.updateRecall([4, 4, 300], 0.6, 1, 16)
ebisu.updateRecall([4, 4, 300], 0.61, 1, 421243)
ebisu.updateRecall([4,4,300], 0.61, 1, 120)
```

Screenshot attached. Running in Safari 15.2 on Mac 12.1, taken from https://fasiha.github.io/ebisu.js/

<img width="1440" alt="Screenshot 2022-04-27 at 11 51 31" src="https://user-images.githubusercontent.com/99540338/165492183-f3100a91-8728-492c-8246-c8d7ff772845.png">

I can dig in an try and diagnose it next week but I was wondering if this was an issue you are familiar with or might have any insight for me?

**Comment by barrelltech** [comment]
Ahh just poked into the source code and it requires whole numbers. However, if I pass values like 61 & 100 I get this error: 

<img width="315" alt="Screenshot 2022-04-27 at 11 54 55" src="https://user-images.githubusercontent.com/99540338/165492828-51b930cf-4248-47d3-9650-ef770c261e6d.png">

```
Error: {"prior":[4,4,300],"successes":61,"total":100,"tnow":120,"rebalance":true,"tback":300,"mean":1.6059075545285275,"m2":0.07552813502953318,"sig2":-2.5034109386622623}
f — ebisu.min.js:21:450
Global Code
evaluateWithScopeExtension
(anonymous function)
_wrapCall
```

Do you have any suggestions on how I might be able to use this library safely with as precise as values as possible?

**Comment by fasiha** [comment]
Sorry for the delay! Yes, it’s a little bit obscure but there’s a pretty low limit on `total` due to numerical instability of evaluating the posterior. I allude to this limit here: https://github.com/fasiha/ebisu/blob/e9149049089a6904ac695a7f02895e5006bf09ab/ebisu/ebisu.py#L105
> N.B. This function is tested for numerical stability for small `total < 5`. It may be unstable for much larger `total`.

The idea behind this feature is really to model apps like Duolingo where, instead of calling `updateRecall` after each “question” you answer, your “questions” are actually more like “sessions” and you update after a series of questions during which the same fact might be reviewed for a `total` of more than one time. In the Duolingo datasets for example this is 2-3 times (if I recall correctly).

It’s alas not able to handle your use case where you’re converting a subjective confidence in your quiz performance to a rational number (100/120, 1/16, etc.). Is this indeed what you’re doing? Let me know if that makes sense or if I’m misunderstanding! Life will be a bit busy for a few more days but I’ll do my best to help out!

**Comment by fasiha** [comment]
The original issue, passing in floating point `successes` is now available in v2.1, just released!

If you run into anything, please feel free to reopen or update!!
